tests/functional/api/test_moderation.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import pytest
+
+# String type for request/response headers and metadata in WSGI.
+#
+# Per PEP-3333, this is intentionally `str` under both Python 2 and 3, even
+# though it has different meanings.
+#
+# See https://www.python.org/dev/peps/pep-3333/#a-note-on-string-types
+native_str = str
+
+
+@pytest.mark.functional
+class TestPutHide(object):
+
+    def test_it_returns_http_204_for_group_creator(self,
+                                                   app,
+                                                   group_annotation,
+                                                   user_with_token):
+        user, token = user_with_token
+        headers = {'Authorization': str('Bearer {}'.format(token.value))}
+
+        res = app.put('/api/annotations/{id}/hide'.format(id=group_annotation.id),
+                                                          headers=headers)
+
+        # The creator of a group has moderation rights over the annotations in that group
+        assert res.status_code == 204
+
+    def test_it_returns_http_404_if_annotation_is_in_world_group(self,
```

Comment:
This test name is a bit misleading because what it is testing is the response if the user lacks moderation rights. The world group is just a group that meets the criteria in this case. The name might imply that the "Public" group is somehow special with respect to moderation, which is not the case.